Meta: Replace IPC linter with something that works

BenWiederhake opened the pull request:
The old `Meta/lint-ipc-ids.sh` script was silently broken in 62af6cd4f9637b8937e59832f5af00f4556c496b. This PR mainly removes the script, and replaces it by a tool that runs similarly to the Markdown checker. Specifically:
- Remove hard-coded magic numbers and the syntax for it (this also simplifies the syntax so that it is easier to parse to name)
- Implement the linter itself
- Run it during CI and pre-commit
- Remove the old script

AtkinsSJ commented:
English nit-pick: You say "replace foo _with_ x" not "replace foo _by_ x". (You would use "by" when it's followed by a -ing verb (gerund?), like "replace foo by generating bars", but that's not common.)

Not important but thought I'd say anyway. :^)

BenWiederhake commented on Meta/Lagom/Tools/CMakeLists.txt:
```diff
@@ -11,5 +11,6 @@ function(lagom_tool tool)
-add_subdirectory(ConfigureComponents)
 add_subdirectory(CodeGenerators)
+add_subdirectory(ConfigureComponents)
+add_subdirectory(IpcMagicLinter)
```
Currently, the naming is all over the place. As the most egregious example, we have both `JSONValue.h` and `JsonValue.h` in the repo. I was hoping to pull the tow in this tug of war a bit in the `JsonValue.h` direction :D

Yes, the names are JSON, HTML, and IPC. And yet I suggest to name the classes JsonValue, HtmlElement, and IpcMagicLinter.

Yes, no, maybe? :)

alimpfard replied:
I'll take that, seems fine - even if "Ipc" is weird and sounds like a prefix of a drug's name.

BenWiederhake commented on Meta/Lagom/Tools/IpcMagicLinter/main.cpp:
```diff
@@ -0,0 +1,93 @@
+    bool had_errors = false;
+    for (int file_index = 1; file_index < argc; ++file_index) {
+        String filename(argv[file_index]);
+        auto file_or_error = Core::File::open(argv[file_index], Core::OpenMode::ReadOnly);
```
Doink, you're right! :D

BenWiederhake commented:
Changes since last push:
- Use a String where the String was already available (thanks Ali!)
- Add new commit that changes how the endpoint magic is treated in gethostbyname and gethostbyaddr
- Rebase for convenience (and to resolve conflict with e.g. c2d44209a8f508ccb243410fc0e591976820b287)
- Make extra sure it really doesn't break:

![Bildschirmfoto_2021-11-03_23-04-10](https://user-images.githubusercontent.com/2690845/140200219-a4319e47-6c2e-4136-a8e2-60209bc4db8a.png)

BenWiederhake commented:
Changes since last push:
- Renamed from IpcMagicLinter to IPCMagicLinter (I object to this naming scheme, but whatever)
- Fixed the copyright declaration in `Meta/Lagom/Tools/IPCMagicLinter/main.cpp` (d'oh!)
- Rebased
